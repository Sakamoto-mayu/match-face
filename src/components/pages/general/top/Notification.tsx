import { twMerge } from 'tailwind-merge';

type MenuProps = {
  title: string;
  completedTasksNumbers: number;
  incompletedTasksNumbers: number;
  deadlineDay: string;
  deadlineNumbers: number;
  className: string;
};

export default function Notification({
  title,
  completedTasksNumbers,
  incompletedTasksNumbers,
  deadlineDay,
  deadlineNumbers,
  className,
}: MenuProps) {
  //buttonのclassNameは親コンポーネントでclassNameを指定して上書き可能
  const mergedClassName = twMerge('p-5 w-1/2 h-auto shadow-lg', className);
  return (
    <>
      <div className={mergedClassName}>
        <h1 className="border-b-[2px] border-b-black shadow-md text-2xl text-left w-full font-bold h-12">
          {title}
        </h1>

        <div className="">
          {/* answer_statusがtrueの件数? */}
          <h2 className="">【対応済】{completedTasksNumbers}件</h2>
          {/* answer_statusがfalseの件数? */}
          <h2 className="pt-4">【未対応】{incompletedTasksNumbers}件</h2>
          {/* データ構造がはっきりしてから編集要*/}
          <ul className=" pl-4">
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
          </ul>
        </div>
      </div>
    </>
  );
}
