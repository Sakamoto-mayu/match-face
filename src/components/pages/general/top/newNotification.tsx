import { twMerge } from 'tailwind-merge';

type MenuProps = {
  completedTasksNumbers: number;
  incompletedTasksNumbers: number;
  deadlineDay: string;
  deadlineNumbers: number;
  className: string;
};

export default function Notification({
  completedTasksNumbers,
  incompletedTasksNumbers,
  deadlineDay,
  deadlineNumbers,
  className,
}: MenuProps) {
  //buttonのclassNameは親コンポーネントでclassNameを指定して上書き可能
  const mergedClassName = twMerge('p-5 w-3/5 h-auto shadow-lg', className);
  return (
    <>
      <div className={mergedClassName}>
        <h1 className="border-b-[2px] border-b-black shadow-md text-2xl text-left w-full font-bold h-12">
          進捗状況
        </h1>

        <div className="pt-5">
          <h2 className="">【対応済】{completedTasksNumbers}件</h2>
          <h2 className="pt-5">【未対応】{incompletedTasksNumbers}件</h2>
          <ul className=" pl-5">
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
          </ul>
        </div>
      </div>
    </>
  );
}
