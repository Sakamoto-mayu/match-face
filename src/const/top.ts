export const MenuContentsArray = [
  {
    id: 1,
    title: '回答する',
    url: '/questions',
    description: '質問に答えて顔合わせの準備をします。',
    imgUrl: '/icon/document_icon.png',
    imgAlt: '回答一覧画面アイコン',
  },
  {
    id: 2,
    title: '回答履歴',
    url: '/histories',
    description: '自身の回答をここから確認してください。',
    imgUrl: '/icon/history_icon.png',
    imgAlt: '回答履歴画面アイコン',
  },
  //   {
  //     id: 3,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
  //   {
  //     id: 4,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
  //   {
  //     id: 5,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
  //   {
  //     id: 6,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
  //   {
  //     id: 7,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
];
