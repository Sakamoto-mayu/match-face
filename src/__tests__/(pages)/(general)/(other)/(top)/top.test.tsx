import Top from '@/app/(pages)/(general)/(other)/(top)/page';
import { render, screen, waitFor } from '@testing-library/react';
import { menuDataMock } from '@/__tests__/components/pages/general/top/top-data-mock';
import { cookies } from 'next/headers';

describe('ユーザートップページのテスト', () => {
  test('fetchが呼ばれていること', async () => {
    global.fetch = jest.fn().mockImplementation(menuDataMock);

    jest.mock('next/headers', () => ({
      cookies: jest.fn(),
    }));
    // cookies.mockImplementation(() => ({ get: () => '1' }));
    render(<Top />);
    // await waitFor(() => {
    expect(fetch).toBeCalledTimes(1);
    // });
  });
});
