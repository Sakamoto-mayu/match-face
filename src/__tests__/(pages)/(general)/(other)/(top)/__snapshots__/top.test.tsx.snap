// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショット スナップショット 1`] = `
<div>
  <a
    href="/questions"
  >
    <button
      class="flex p-5 w-96 h-40 shadow-lg hover:bg-gray-100 flex-col active:bg-gray-200"
    >
      <p
        class="border-b-[1px] border-b-black shadow-md text-2xl text-left w-full font-bold"
      >
        回答する
      </p>
      <div
        class="flex pt-4 w-full"
      >
        <p
          class="text-left mr-auto"
        >
          質問に答えて顔合わせの準備をします。
        </p>
        <img
          alt="回答一覧画面アイコン"
          class="pt-2 pl-2"
          data-nimg="1"
          decoding="async"
          height="70"
          loading="lazy"
          src="/_next/image?url=%2Ficon%2Fdocument_icon.png&w=256&q=75"
          srcset="/_next/image?url=%2Ficon%2Fdocument_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fdocument_icon.png&w=256&q=75 2x"
          style="color: transparent;"
          width="70"
        />
      </div>
    </button>
  </a>
</div>
`;

exports[`スナップショット スナップショット 2`] = `
<div>
  <div
    class="p-5 w-3/5 h-auto shadow-lg"
  >
    <h1
      class="border-b-[2px] border-b-black shadow-md text-2xl text-left w-full font-bold h-12"
    >
      進捗状況
    </h1>
    <div
      class="pt-5"
    >
      <h2>
        【対応済】
        3
        件
      </h2>
      <h2
        class="pt-5"
      >
        【未対応】
        4
        件
      </h2>
      <ul
        class="pl-5"
      >
        <li
          class="pt-2"
        >
          2023年10月01日
          までに対応必須の項目が
          2
          件あります
        </li>
      </ul>
      <ul
        class="pl-5"
      >
        <li
          class="pt-2"
        >
          2023年10月15日
          までに対応必須の項目が
          1
          件あります
        </li>
      </ul>
      <ul
        class="pl-5"
      >
        <li
          class="pt-2"
        >
          2023年12月15日
          までに対応必須の項目が
          1
          件あります
        </li>
      </ul>
    </div>
  </div>
</div>
`;
